client/CPlayerInterface.cpp

```diff
@@ -1013,7 +1013,7 @@ void CPlayerInterface::showInfoDialog(const std::string &text, const std::vector
 	}
 	std::shared_ptr<CInfoWindow> temp = CInfoWindow::create(text, playerID, components);
 
-	if (makingTurn && GH.windows().count() > 0 && LOCPLINT == this)
+	if ((makingTurn || battleInt) && GH.windows().count() > 0 && LOCPLINT == this)
```

Comment:
Can you check this version?
```suggestion
	if ((makingTurn || battleInt->curInt.get() == this) && GH.windows().count() > 0 && LOCPLINT == this)
```

Reply:
Sure. Currently AFK. Will fix in about 1 hr

Reply:
It works with some additional checks that `battleInt` and `battleInt->curInt` are not null. Otherwise exceptions are thrown.